linux/SystemdMeter.c

```diff
@@ -221,16 +265,30 @@ static void updateViaExec(void) {
       close(fdnull);
       // Use of NULL in variadic functions must have a pointer cast.
       // The NULL constant is not required by standard to have a pointer type.
-      execlp(
-         "systemctl",
-         "systemctl",
-         "show",
-         "--property=SystemState",
-         "--property=NFailedUnits",
-         "--property=NNames",
-         "--property=NJobs",
-         "--property=NInstalledJobs",
-         (char *)NULL);
+      if (user) {
+         execlp(
+            "systemctl",
+            "systemctl",
+            "show",
+            "--user",
+            "--property=SystemState",
+            "--property=NFailedUnits",
+            "--property=NNames",
+            "--property=NJobs",
+            "--property=NInstalledJobs",
+            (char *)NULL);
+      } else {
+         execlp(
+            "systemctl",
+            "systemctl",
+            "show",
+            "--property=SystemState",
+            "--property=NFailedUnits",
+            "--property=NNames",
+            "--property=NJobs",
+            "--property=NInstalledJobs",
+            (char *)NULL);
+      }
```

Comment:
Given that there's an implied default `--system`, we can simplify here. :)
```suggestion
      execlp(
         "systemctl",
         "systemctl",
         "show",
         user ? "--user" : "--system",
         "--property=SystemState",
         "--property=NFailedUnits",
         "--property=NNames",
         "--property=NJobs",
         "--property=NInstalledJobs",
         (char *)NULL);
```